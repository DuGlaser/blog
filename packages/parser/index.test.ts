import renderer from 'react-test-renderer';

import { parser } from './index';

describe('parser', () => {
  test('# test', () => {
    const result = parser('# test');
    const component = renderer.create(result);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('## test', () => {
    const result = parser('## test');
    const component = renderer.create(result);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('> test', () => {
    const result = parser('> test');
    const component = renderer.create(result);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('- test', () => {
    const result = parser('- test');
    const component = renderer.create(result);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('* test', () => {
    const result = parser('* test');
    const component = renderer.create(result);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('code', () => {
    const result = parser(`
                          \`\`\`js
                            console.log('test')
                          \`\`\`
                          `);
    const component = renderer.create(result);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('break line', () => {
    const result = parser(`hoge
                           hoge`);
    const component = renderer.create(result);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('[example](http://example.com)', () => {
    const result = parser('[example](http://example.com)');
    const component = renderer.create(result);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
