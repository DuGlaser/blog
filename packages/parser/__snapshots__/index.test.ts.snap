// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser # test 1`] = `
<div>
  <h1>
    test
  </h1>
</div>
`;

exports[`parser ## test 1`] = `
<div>
  <h2>
    test
  </h2>
</div>
`;

exports[`parser * test 1`] = `
<div>
  <ul>
    

    <li>
      test
    </li>
    

  </ul>
</div>
`;

exports[`parser > test 1`] = `
<div>
  <blockquote>
    

    <p>
      test
    </p>
    

  </blockquote>
</div>
`;

exports[`parser [example](http://example.com) 1`] = `
<div>
  <p>
    <a
      href="http://example.com"
    >
      example
    </a>
  </p>
</div>
`;

exports[`parser - test 1`] = `
<div>
  <ul>
    

    <li>
      test
    </li>
    

  </ul>
</div>
`;

exports[`parser break line 1`] = `
<div>
  <p>
    hoge
    <br />
    

    hoge
  </p>
</div>
`;

exports[`parser code 1`] = `
<div>
  <pre>
    <code>
                            \`\`\`js
                        console.log('test')
                      \`\`\`
                      

    </code>
  </pre>
</div>
`;
